### PR TITLE
TST: remove patches to pandas.util.testing.N

### PR DESCRIPTION
**Pull request by simonjayhawkins**
xref https://github.com/pandas-dev/pandas/pull/24769#issuecomment-455326819

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Another solution might be to add this as a keyword argument to those functions, so the monkeypatch is not needed?

**Comment by simonjayhawkins** [comment]
@jorisvandenbossche 

> so the monkeypatch is not needed?

are you not a fan of monkeypatch?

there are only 4 instances where the patch is needed, one of those is for depecated Panel another instance is code duplication resulting from some tests using unittest setup method and other tests having the setup method converted to pytest fixtures.

the monkeypatch used as context manager is to make it more explicit what is happening and ensuring that the patch in the fixture is undone before the test executes. this may appear to add additional lines of code.

my only concern with altering pandas.util.testing is that it is in the pandas directory and not the tests directory, so a change here would be considered an api change?

File: pandas/tests/indexing/multiindex/conftest.py [hunk 1]
```diff
@@ -18,11 +18,12 @@ def multiindex_dataframe_random_data():
 
 
 @pytest.fixture
-def multiindex_year_month_day_dataframe_random_data():
+def multiindex_year_month_day_dataframe_random_data(monkeypatch):
     """DataFrame with 3 level MultiIndex (year, month, day) covering
     first 100 business days from 2000-01-01 with random data"""
-    tm.N = 100
-    tdf = tm.makeTimeDataFrame()
+    with monkeypatch.context() as m:
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/conftest.py]: why do all this, ``makeTimeDataFrame(100)`` will do this

**Review by jreback — CHANGES_REQUESTED** [review]
see my comments; all of the examples should not need patching

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
looks ok, ping on green.

**Comment by simonjayhawkins** [comment]
@jreback green!

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @simonjayhawkins

**Pull request merged**